src/main/java/dev/seniorjava/githubstatisticsbot/stats/collectors/ReactionStatisticsCollector.java

```diff
@@ -0,0 +1,76 @@
+package dev.seniorjava.githubstatisticsbot.stats.collectors;
+
+import dev.seniorjava.githubstatisticsbot.exceptions.CannotFetchGithubDataException;
+import dev.seniorjava.githubstatisticsbot.stats.mappers.StatisticsMapper;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ReactionStatisticsCollector {
+
+  private final OkHttpClient httpClient;
+  private final StatisticsMapper mapper;
+
+  public Collection<ReactionStatistics> fromPullRequest(final String sourceURL) {
+    final String targetURL = getPullRequestReactionsURL(sourceURL);
+    return collectStats(targetURL);
+  }
+
+  public Collection<ReactionStatistics> fromCommit(final String sourceURL) {
+    final String targetURL = getCommitReactionsURL(sourceURL);
+    return collectStats(targetURL);
+  }
+
+  private Collection<ReactionStatistics> collectStats(final String targetURL) {
+    try {
+      final String jsonContent = getJsonContent(targetURL);
+      return mapper.map(jsonContent);
+    } catch (Exception exception) {
+      log.error("Cannot fetch pull request.", exception);
+    }
+    return Collections.emptyList();
+  }
+
+  private String getJsonContent(final String targetURL) throws IOException {
+    final Response response = httpClient.newCall(createRequest(targetURL)).execute();
+    final ResponseBody body = response.body();
+    if (response.isSuccessful() && body != null) {
+      final String json = body.string();
+      body.close();
+      return json;
+    }
+    response.close();
```

Comment:
Класс `Response` [реализует](https://square.github.io/okhttp/3.x/okhttp/index.html?okhttp3/Response.html) `AutoClosable`, поэтому лучше обернуть всю операцию в `try-with-resources` блок, но не закрывать вручную:
```suggestion
    try (final var response = httpClient.newCall(createRequest(targetURL)).execute()) {}
```

Reply:
Didn't pay attention to this. This is a very good suggestion. Thank you 🤝